test/gtest/uct/ib/test_dc.cc

```diff
@@ -440,6 +440,32 @@ UCS_TEST_P(test_dc, rand_dci_pending_purge) {
     flush();
 }
 
+UCS_TEST_P(test_dc, pool_indices)
+{
+    constexpr uint8_t max_dci_pools = UCT_DC_MLX5_IFACE_MAX_DCI_POOLS;
+    entity *rand_e;
+    uint8_t num_dci_pools;
+    uct_dc_mlx5_ep_t *ep;
+    uint8_t expected_pool_index;
+    int i;
+
+    /* Set num paths to max */
+    modify_config("IB_NUM_PATHS", std::to_string(max_dci_pools).c_str());
+
+    rand_e        = create_rand_entity();
+    num_dci_pools = dc_iface(rand_e)->tx.num_dci_pools;
+
+    EXPECT_EQ(num_dci_pools, max_dci_pools);
+
+    /* Create an ep for each pool */
+    for (i = 0; i < max_dci_pools; i++) {
+        rand_e->connect_to_iface(i, *m_e2, i);
+        expected_pool_index = i % num_dci_pools;
```

Comment:
maybe expected_pool_index could be the current pool index, before the send operation, and not necessarily assume that pool assignment is round robin?

Reply:
I'm not sure what you mean by:
> the current pool index, before the send operation

the pool index is being set internally on ep creation so it will be like comparing pool_index(ep) with itself, no?

Reply:
I see.. any way to change the test to not expect specific pool index?
maybe check that pool index is in a given range?

Reply:
what happens if there are not enough bits?
maybe we can check at least one DCI uses the last pool?